client/main.lua

```diff
@@ -40,53 +31,48 @@ RegisterNetEvent('QBCore:Client:OnJobUpdate', function(JobInfo)
 end)
 
 function hintToDisplay(text)
-    SetTextComponentFormat("STRING")
-    AddTextComponentString(text)
-    DisplayHelpTextFromStringLabel(0, 0, 1, 20000)
+	exports['qb-core']:DrawText(text)
+end
+
+function hideLastHint()
+	exports['qb-core']:HideText()
 end
 
 -- Ped spawn and mission accept
 Citizen.CreateThread(function()
-    while true do
-        Citizen.Wait(0)
-        local plyCoords = GetEntityCoords(PlayerPedId(), false)
-        local dist = #(plyCoords - vector3(MissionMarker.x, MissionMarker.y, MissionMarker.z))
-
-        if dist <= 25.0 then
-            if not DoesEntityExist(dealer) then
-                RequestModel("s_m_y_dealer_01")
-                while not HasModelLoaded("s_m_y_dealer_01") do
-                    Wait(10)
-                end
-                dealer = CreatePed(26, "s_m_y_dealer_01", dealerCoords.x, dealerCoords.y, dealerCoords.z, 268.9422,
-                    false, false)
-                SetEntityHeading(dealer, 1.8)
-                SetBlockingOfNonTemporaryEvents(dealer, true)
-                TaskStartScenarioInPlace(dealer, "WORLD_HUMAN_AA_SMOKE", 0, false)
-            end
-            DrawMarker(25, MissionMarker.x, MissionMarker.y, MissionMarker.z - 0.90, 0, 0, 0, 0, 0, 0, 1.301, 1.3001, 1.3001, 0, 205, 250, 200, 0, 0, 0, 0)
-        else
-            Citizen.Wait(1500)
-        end
-
-        if dist <= 1.0 then
-            QBCore.Functions.DrawText3D(MissionMarker.x, MissionMarker.y, MissionMarker.z,
-                "~g~[E]~b~ To accept missions")
-            if IsControlJustPressed(0, 38) then
-                TriggerServerEvent("AttackTransport:akceptujto")
-                Citizen.Wait(500)
-            end
-        end
-
-    end
+	while true do
+		local plyCoords = GetEntityCoords(PlayerPedId(), false)
+		local dist = #(plyCoords - vector3(Config.MissionMarker.x, Config.MissionMarker.y, Config.MissionMarker.z))
+		if dist <= 25.0 then
+			if not DoesEntityExist(dealer) then
+				RequestModel("s_m_y_dealer_01")
+				while not HasModelLoaded("s_m_y_dealer_01") do
+					Wait(10)
+				end
+				dealer = CreatePed(26, "s_m_y_dealer_01", Config.DealerCoords.x, Config.DealerCoords.y, Config.DealerCoords.z, 268.9422, false, false)
+				SetEntityHeading(dealer, 1.8)
+				SetBlockingOfNonTemporaryEvents(dealer, true)
+				TaskStartScenarioInPlace(dealer, "WORLD_HUMAN_AA_SMOKE", 0, false)
+			end
+			if dist <= 2.0 then
+				QBCore.Functions.DrawText3D(Config.MissionMarker.x, Config.MissionMarker.y, Config.MissionMarker.z, "~b~[E]~w~ To accept mission")
+				if IsControlJustPressed(0, 38) then
+					TriggerServerEvent("AttackTransport:akceptujto")
+					Citizen.Wait(500)
+				end
+			end
+		else
```

Comment:
Instead of this else statement, add a sleep variable and change the sleep to 0 when the dist is under 25.0 but make the default value of sleep 500 so it changes like that